src/DataLayouts/broadcast.jl

```diff
@@ -3,6 +3,11 @@
 
 abstract type DataStyle <: Base.BroadcastStyle end
 
+abstract type Data0DStyle <: DataStyle end
+struct DataFStyle{A} <: Data0DStyle end
+DataStyle(::Type{DataF{S, A}}) where {S, A} = DataFStyle{parent_array_type(A)}()
+Data0DStyle(::Type{DataFStyle{A}}) where {A} = DataFStyle{A}
```

Comment:
@simonbyrne when does this `Data0DStyle` need to be defined for other data styles? I'm looking at 
```
DataColumnStyle(::Type{VIFHStyle{Ni, A}}) where {Ni, A} = VFStyle{A}
DataSlab1DStyle(::Type{VIFHStyle{Ni, A}}) where {Ni, A} = IFStyle{Ni, A}
```
and wondering when the analogous methods are needed in this case.